### PR TITLE
NAS-105128 / 12.0 / NAS-105128 Add stop services to pool export dialog

### PR DESCRIPTION
**Pull request by dkmullen**

### CODE REVIEW

File: src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts [hunk 1]
```diff
@@ -598,17 +598,28 @@ export class VolumesListTableConfig implements InputTableConf {
...
+                      conditionalErrMessage += '<div class="warning-box">These services must be stopped to export the pool:';
+                      res.exc_info.extra.stop_services.forEach((item) => {
+                        conditionalErrMessage += `<br>- ${item}`;
+                      });
+                    }
+                    if (res.exc_info.extra.restart_services.length > 0) {
+                      if (res.exc_info.extra.stop_services.length > 0) {
+                        conditionalErrMessage += '<br><br>';
+                      }
+                      conditionalErrMessage += '<div class="warning-box">These services must be restarted to export the pool:';
```
Comment (erinix) [inline src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts]: You should pipe this string to translate

File: src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts [hunk 2]
```diff
@@ -598,17 +598,28 @@ export class VolumesListTableConfig implements InputTableConf {
...
                   if (res.exc_info.extra && res.exc_info.extra['code'] === 'control_services') {
                     entityDialog.dialogRef.close(true);
                     dialogRef.close(true);
-                    conditionalErrMessage = '<div class="warning-box">Warning: These services must be restarted to export the pool:<br>';
-                    res.exc_info.extra.services.forEach((item) => {
-                      conditionalErrMessage += `<br>- ${item}`;
-                    })
-                    conditionalErrMessage += '<br><br>Exporting/disconnecting will continue after services have been restarted.</div><br />';
+                    if (res.exc_info.extra.stop_services.length > 0) {
+                      conditionalErrMessage += '<div class="warning-box">These services must be stopped to export the pool:';
```
Comment (erinix) [inline src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts]: this one too

File: src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts [hunk 3]
```diff
@@ -598,17 +598,28 @@ export class VolumesListTableConfig implements InputTableConf {
...
+                    if (res.exc_info.extra.restart_services.length > 0) {
+                      if (res.exc_info.extra.stop_services.length > 0) {
+                        conditionalErrMessage += '<br><br>';
+                      }
+                      conditionalErrMessage += '<div class="warning-box">These services must be restarted to export the pool:';
+                      res.exc_info.extra.restart_services.forEach((item) => {
+                        conditionalErrMessage += `<br>- ${item}`;
+                      });
+                    }
+                    conditionalErrMessage += '<br><br>Exporting/disconnecting will continue after services have been handled.</div><br />';
```
Comment (Mrt134) [inline src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts]:
```suggestion
                    conditionalErrMessage += '<br><br>Exporting/disconnecting will continue after affected services have been managed.</div><br />';
```

File: src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts [hunk 4]
```diff
@@ -598,17 +598,28 @@ export class VolumesListTableConfig implements InputTableConf {
...
                       self.dialogService.confirm(T("Error exporting/disconnecting pool."),
-                        conditionalErrMessage, true, 'Restart Services and Continue')
+                        conditionalErrMessage, true, 'Handle Services and Continue')
```
Comment (Mrt134) [inline src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts]:
```suggestion
                        conditionalErrMessage, true, 'Manage Services and Continue')
```

**Pull request merged**